cpp/models/ide_secir/model.cpp

```diff
@@ -195,49 +250,93 @@ void Model::compute_flow(int idx_InfectionTransitions, Eigen::Index idx_Incoming
 
 void Model::flows_current_timestep(ScalarType dt)
 {
-    // Calculate flow from S to E with force of infection from previous time step und susceptibles from current time step.
+    // Calculate flow SusceptibleToExposed with the force of infection from previous time step und susceptibles from current time step.
     m_transitions.get_last_value()[Eigen::Index(InfectionTransition::SusceptibleToExposed)] =
         dt * m_forceofinfection * m_populations.get_last_value()[Eigen::Index(InfectionState::Susceptible)];
+
     // Calculate all other flows with compute_flow.
-    // Flow from E to C
+    // Flow ExposedToInfectedNoSymptoms.
     compute_flow((int)InfectionTransition::ExposedToInfectedNoSymptoms,
                  Eigen::Index(InfectionTransition::SusceptibleToExposed), dt);
-    // Flow from C to I
+    // Flow InfectedNoSymptomsToInfectedSymptoms.
     compute_flow((int)InfectionTransition::InfectedNoSymptomsToInfectedSymptoms,
                  Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms), dt);
-    // Flow from C to R
+    // Flow InfectedNoSymptomsToRecovered.
     compute_flow((int)InfectionTransition::InfectedNoSymptomsToRecovered,
                  Eigen::Index(InfectionTransition::ExposedToInfectedNoSymptoms), dt);
-    // Flow from I to H
+    // Flow InfectedSymptomsToInfectedSevere.
     compute_flow((int)InfectionTransition::InfectedSymptomsToInfectedSevere,
                  Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms), dt);
-    // Flow from I to R
+    // Flow InfectedSymptomsToRecovered.
     compute_flow((int)InfectionTransition::InfectedSymptomsToRecovered,
                  Eigen::Index(InfectionTransition::InfectedNoSymptomsToInfectedSymptoms), dt);
-    // Flow from H to U
+    // Flow InfectedSevereToInfectedCritical.
     compute_flow((int)InfectionTransition::InfectedSevereToInfectedCritical,
                  Eigen::Index(InfectionTransition::InfectedSymptomsToInfectedSevere), dt);
-    // Flow from to H to R
+    // Flow InfectedSevereToRecovered.
     compute_flow((int)InfectionTransition::InfectedSevereToRecovered,
                  Eigen::Index(InfectionTransition::InfectedSymptomsToInfectedSevere), dt);
-    // Flow from U to D
+    // Flow InfectedCriticalToDead.
     compute_flow((int)InfectionTransition::InfectedCriticalToDead,
                  Eigen::Index(InfectionTransition::InfectedSevereToInfectedCritical), dt);
-    // Flow from U to R
+    // Flow InfectedCriticalToRecovered.
     compute_flow((int)InfectionTransition::InfectedCriticalToRecovered,
                  Eigen::Index(InfectionTransition::InfectedSevereToInfectedCritical), dt);
 }
 
-void Model::compute_deaths()
+void Model::update_compartments_current_timestep()
```

Comment:
**Important** Why is susceptible not reduced here???

Reply:
This is because we need the Susceptibles of the current time step to calculate the SusceptibleToExposed transition. Therefore, S must be calculated before the transitions. To update the other compartments, we need the transitions of the current time step, so the compartments have to be calculated after the transitions. This means that the calculation order in the simulation function cannot be changed and S cannot be calculated with the other compartments!